spec/unit/facter/udev_spec.rb

```diff
@@ -0,0 +1,163 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe 'udev fact', type: :fact do
+  subject(:fact) { Facter.fact(:udev) }
+
+  before { Facter.clear }
+
+  context 'when kernel is linux and udevadm is present' do
+    before do
+      allow(Facter.fact(:kernel)).to receive(:value).and_return(:linux)
+      allow(Facter.fact(:udevadm)).to receive(:value).and_return('path' => '/foo')
+      allow(Facter::Util::Resolution).to receive(:exec).with('udevadm info -e').and_return(
+        File.read(fixtures('udevdb', udevdb_fixture))
+      )
+    end
+
+    context 'centos8-supermicro-1114S-WN10RT' do
+      let(:udevdb_fixture) { 'centos8-supermicro-1114S-WN10RT.txt' }
+
+      it do
+        expect(Facter.value(:udev)).to include(
```

Comment:
Is this a better pattern?
```suggestion
        expect(subject.value).to include(
```
Or since it's named
```suggestion
        expect(fact.value).to include(
```

Reply:
Yes, that is definitely an improvement.

Reply:
Done.